### PR TITLE
Set small DefaultJobPodFailureBackOff in Job integration tests

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

To make the Job integration tests faster. For example, on my testing environment:
- `TestNonParallelJob` from 16s to 7s
- `TestJobPodFailurePolicy` from 58s to 28s
- `TestParallelJobWithCompletions` from 72s to 38s
- `TestIndexedJob` from 12s to 9s

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

I set the default backoff small for pod failures in these tests that it was making a difference.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/job/job_test.go [hunk 1]
```diff
@@ -737,6 +739,7 @@ func TestNonParallelJob(t *testing.T) {
 }
 
 func TestParallelJob(t *testing.T) {
+	t.Cleanup(setDurationDuringTest(&jobcontroller.DefaultJobPodFailureBackOff, time.Millisecond))
```
[the remark this answers is not in the document]
Comment (mimowo) [reply test/integration/job/job_test.go]: by mistake, fixed.

File: test/integration/job/job_test.go [hunk 2]
```diff
@@ -467,6 +467,7 @@ func TestJobPodFailurePolicyWithFailedPodDeletedDuringControllerRestart(t *testi
 // TestJobPodFailurePolicy tests handling of pod failures with respect to the
 // configured pod failure policy rules
 func TestJobPodFailurePolicy(t *testing.T) {
+	t.Cleanup(setDurationDuringTest(&jobcontroller.DefaultJobPodFailureBackOff, 100*time.Millisecond))
```
Comment (alculquicondor) [inline test/integration/job/job_test.go:470]: Let's add a constant `fastPodFailureBackoff`

Comment (mimowo) [reply test/integration/job/job_test.go]: Done

**Pull request merged**